<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting Started • EpiProcess</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started">
<meta property="og:description" content="EpiProcess">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EpiProcess</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/EpiProcess.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/EpiProcess/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting Started</h1>
                        <h4 data-toc-skip class="author"><h4>
Authors: <i>Brian Schilder, Leyla Abbasova</i>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
Vignette updated: <i>Mar-09-2022</i>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/EpiProcess/blob/HEAD/vignettes/EpiProcess.Rmd" class="external-link"><code>vignettes/EpiProcess.Rmd</code></a></small>
      <div class="hidden name"><code>EpiProcess.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/EpiProcess" class="external-link">EpiProcess</a></span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">root</span><span class="op">=</span><span class="st">"/Volumes/bms20/projects/neurogenomics-lab/live/Data/tip_seq"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">root</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "/Volumes/bms20/projects/neurogenomics-lab/live/Data/tip_seq"</span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">datadir</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span>
  <span class="va">root</span>,<span class="st">"raw_data/H3K27ac_K562_cells/phase_1_05_jan_2022/X204SC21121860-Z01-F001/raw_data"</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">datadir</span><span class="op">)</span> </code></pre></div>
<pre><code><span class="co">## [1] "/Volumes/bms20/projects/neurogenomics-lab/live/Data/tip_seq/raw_data/H3K27ac_K562_cells/phase_1_05_jan_2022/X204SC21121860-Z01-F001/raw_data"</span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fastq_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">datadir</span>, pattern <span class="op">=</span> <span class="st">".fastq.gz|.fq.gz"</span>,
                          full.names <span class="op">=</span> <span class="cn">TRUE</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="fu">echoconda</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/echoconda/man/yaml_to_env.html" class="external-link">yaml_to_env</a></span><span class="op">(</span><span class="st">"epiprocess"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## echoconda:: Conda already installed.</span></code></pre>
<pre><code><span class="co">## Identified yaml file stored in echoconda.</span></code></pre>
<pre><code><span class="co">## Yaml contents:</span></code></pre>
<pre><code><span class="co">## name: epiprocess</span>
<span class="co">## channels:</span>
<span class="co">##   - conda-forge</span>
<span class="co">##   - bioconda</span>
<span class="co">##   - nodefaults</span>
<span class="co">## dependencies:</span>
<span class="co">##   # Python</span>
<span class="co">##   - python&gt;=3.6.1</span>
<span class="co">##   - pandas&gt;=0.25.0</span>
<span class="co">##   # Command line</span>
<span class="co">##   - fastqc</span>
<span class="co">##   - trim-galore</span>
<span class="co">##   - bowtie2</span>
<span class="co">##   - picard</span>
<span class="co">##   - samtools</span>
<span class="co">##   - bedtools</span>
<span class="co">##   - seacr</span>
<span class="co">##   - deeptools</span>
<span class="co">##   - macs2</span>
<span class="co">##   # R</span>
<span class="co">##   - r&gt;=4.0.0</span>
<span class="co">##   - r-devtools</span>
<span class="co">##   - r-patchwork</span>
<span class="co">##   - r-gert</span>
<span class="co">##   - r-r.oo</span>
<span class="co">##   - r-data.table</span>
<span class="co">##   - r-ggplot2</span>
<span class="co">##   - r-tidyverse</span>
<span class="co">##   # R: Bioconductor</span>
<span class="co">##   - r-biocmanager</span>
<span class="co">##   - radian</span>
<span class="co">##   - pip</span></code></pre>
<pre><code><span class="co">## </span></code></pre>
<pre><code><span class="co">## echoconda:: Conda environment already exists: epiprocess</span></code></pre>
<pre><code><span class="co">## Time difference of 0.3 secs</span></code></pre>
<pre><code><span class="co">## [1] "epiprocess"</span></code></pre>
<div class="section level2">
<h2 id="pre-processing">Pre-processing<a class="anchor" aria-label="anchor" href="#pre-processing"></a>
</h2>
<div class="section level3">
<h3 id="fastqc">FastQC<a class="anchor" aria-label="anchor" href="#fastqc"></a>
</h3>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a> </span></code></pre></div>
</div>
<div class="section level3">
<h3 id="adapter-trimming">Adapter trimming<a class="anchor" aria-label="anchor" href="#adapter-trimming"></a>
</h3>
<p>Get bad gzip error if use zipped files, so gunzip first.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Linux command</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="va">sampledir</span><span class="op">=</span><span class="st">"rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/raw_data/HKTCKBBXY/IGFQ001138_hu_2-3-2021_CutandTag/IGF117517"</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="va">outdir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY/data_trimmed"</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir -p $outdir</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="va">sample_name</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co">## need cutadapt</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load anaconda3/personal</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate cutadaptenv</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a><span class="ex">/rds/general/user/la420/home/TrimGalore-0.6.6/trim_galore</span> <span class="at">-o</span> <span class="va">$outdir</span> <span class="at">--gzip</span> <span class="at">--basename</span> <span class="va">$sample_name</span> <span class="at">--paired</span> <span class="va">$sampledir</span>/<span class="va">${sample_name}</span>_R1_001.fastq <span class="va">$sampledir</span>/<span class="va">${sample_name}</span>_R2_001.fastq</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="alignment">Alignment<a class="anchor" aria-label="anchor" href="#alignment"></a>
</h2>
<p>Index and other genome files obtained at <a href="https://emea.support.illumina.com/sequencing/sequencing_software/igenome.html" class="external-link uri">https://emea.support.illumina.com/sequencing/sequencing_software/igenome.html</a></p>
<ul>
<li>If adapter trimming is not required (e.g. in case of shorter-read
sequencing, such as 25x25 paired-end sequencing), use the –end-to-end
instead of –local bowtie2 setting. –local is used to avoid remaining
adapter sequence at the 3’ ends of reads during mapping.</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Linux command</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="va">datadir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY"</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="va">ref</span><span class="op">=</span><span class="st">"/rds/general/user/la420/home/hg19/index/genome"</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="va">cores</span><span class="op">=</span>8</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="va">sample_name</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir -p ${datadir}/alignment/sam/bowtie2_summary</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir -p ${datadir}/alignment/bam</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir -p ${datadir}/alignment/bed</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir -p ${datadir}/alignment/bedgraph</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="va">outdir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY/alignment/sam"</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load bowtie2/2.2.9</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a><span class="ex">bowtie2</span> <span class="dt">\</span></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>--local <span class="dt">\ </span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a><span class="ex">--very-sensitive</span> <span class="dt">\</span></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>--no-mixed <span class="dt">\</span></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>--no-discordant <span class="dt">\</span></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>--phred33 <span class="dt">\</span></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>-I 10 <span class="dt">\</span></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>-X 700 <span class="dt">\</span></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>-p <span class="va">${cores}</span> <span class="dt">\</span></span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>-x <span class="va">${ref}</span> <span class="dt">\</span></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>-1 <span class="va">${datadir}</span>/data_trimmed/<span class="va">${sample_name}</span>_val_1.fq.gz <span class="dt">\</span></span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>-2 <span class="va">${datadir}</span>/data_trimmed/<span class="va">${sample_name}</span>_val_2.fq.gz <span class="dt">\</span></span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>-S <span class="va">${outdir}</span>/<span class="va">${sample_name}</span>_trimmed_bowtie2.sam <span class="op">&amp;&gt;</span> <span class="va">${outdir}</span>/bowtie2_summary/<span class="va">${sample_name}</span>_trimmed_bowtie2.txt</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="remove-duplicates">Remove duplicates<a class="anchor" aria-label="anchor" href="#remove-duplicates"></a>
</h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Linux command</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load picard/2.6.0</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY"</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="va">picard_command</span><span class="op">=</span><span class="st">"java -Xmx12G -XX:ParallelGCThreads=5 -XX:GCTimeLimit=50 -jar /apps/picard/2.6.0/picard.jar"</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="va">sample_name</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir -p $workdir/removeDuplicate</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir -p $workdir/removeDuplicate/picard_summary</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a><span class="co">## sort by coordinate</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="va">$picard_command</span> SortSam I=<span class="va">$workdir</span>/alignment/sam/<span class="va">${sample_name}</span>_trimmed_bowtie2.sam O=<span class="va">$workdir</span>/alignment/sam/<span class="va">${sample_name}</span>_trimmed_bowtie2.sorted.sam SORT_ORDER=coordinate</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a><span class="co">## mark duplicates</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a><span class="va">$picard_command</span> MarkDuplicates I=<span class="va">$workdir</span>/alignment/sam/<span class="va">${sample_name}</span>_trimmed_bowtie2.sorted.sam O=<span class="va">$workdir</span>/removeDuplicate/<span class="va">${sample_name}</span>_trimmed_bowtie2.sorted.dupMarked.sam METRICS_FILE=<span class="va">$workdir</span>/removeDuplicate/picard_summary/<span class="va">${sample_name}</span>_picard.dupMark.txt</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a><span class="co">## remove duplicates</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a><span class="va">$picard_command</span> MarkDuplicates I=<span class="va">$workdir</span>/alignment/sam/<span class="va">${sample_name}</span>_trimmed_bowtie2.sorted.sam O=<span class="va">$workdir</span>/removeDuplicate/<span class="va">${sample_name}</span>_trimmed_bowtie2.sorted.rmDup.sam REMOVE_DUPLICATES=true METRICS_FILE=<span class="va">$workdir</span>/removeDuplicate/picard_summary/<span class="va">${sample_name}</span>_picard.rmDup.txt</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="obtain-fragment-size-distribution">Obtain fragment size distribution<a class="anchor" aria-label="anchor" href="#obtain-fragment-size-distribution"></a>
</h2>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Linux command</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load samtools/1.3.1</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY"</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="va">sample_name</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir -p $workdir/alignment/sam/fragmentLen</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="co">## extract the 9th column from the alignment sam file which is the fragment length</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-F</span> 0x04 <span class="va">$workdir</span>/removeDuplicate/<span class="va">${sample_name}</span>_trimmed_bowtie2.sorted.rmDup.sam <span class="kw">|</span> <span class="fu">awk</span> <span class="at">-F</span><span class="st">'\t'</span> <span class="st">'function abs(x){return ((x &lt; 0.0) ? -x : x)} {print abs($9)}'</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="kw">|</span> <span class="fu">uniq</span> <span class="at">-c</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="at">-v</span> OFS=<span class="st">"\t"</span> <span class="st">'{print $2, $1/2}'</span> <span class="op">&gt;</span><span class="va">$workdir</span>/alignment/sam/fragmentLen/<span class="va">${sample_name}</span>_rmDup_fragmentLen.txt</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="convert-files">Convert files<a class="anchor" aria-label="anchor" href="#convert-files"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Linux command</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load samtools/1.3.1</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load bedtools/2.25.0</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY"</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="va">sample_name</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="co">## sort sam file by read name (rather than coordinate, as it is now)</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> sort <span class="at">-n</span> <span class="va">$workdir</span>/removeDuplicate/<span class="va">${sample_name}</span>_trimmed_bowtie2.sorted.rmDup.sam <span class="at">-o</span> <span class="va">$workdir</span>/alignment/sam/<span class="va">${sample_name}</span>_trimmed_bowtie2.n_sorted.rmDup.sam</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a><span class="co">## filter and keep the mapped read pairs</span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-bS</span> <span class="at">-F</span> 0x04 <span class="va">$workdir</span>/alignment/sam/<span class="va">${sample_name}</span>_trimmed_bowtie2.n_sorted.rmDup.sam <span class="op">&gt;</span><span class="va">$workdir</span>/alignment/bam/<span class="va">${sample_name}</span>_bowtie2_rmDup.mapped.bam</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a><span class="co">## convert into bed file format</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a><span class="ex">bedtools</span> bamtobed <span class="at">-i</span> <span class="va">$workdir</span>/alignment/bam/<span class="va">${sample_name}</span>_bowtie2_rmDup.mapped.bam <span class="at">-bedpe</span> <span class="op">&gt;</span><span class="va">$workdir</span>/alignment/bed/<span class="va">${sample_name}</span>_bowtie2_rmDup.bed</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a><span class="co">## keep the read pairs that are on the same chromosome and fragment length less than 1000bp.</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'$1==$4 &amp;&amp; $6-$2 &lt; 1000 {print $0}'</span> <span class="va">$workdir</span>/alignment/bed/<span class="va">${sample_name}</span>_bowtie2_rmDup.bed <span class="op">&gt;</span><span class="va">$workdir</span>/alignment/bed/<span class="va">${sample_name}</span>_bowtie2_rmDup.clean.bed</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a><span class="co">## only extract the fragment related columns</span></span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a><span class="fu">cut</span> <span class="at">-f</span> 1,2,6 <span class="va">$workdir</span>/alignment/bed/<span class="va">${sample_name}</span>_bowtie2_rmDup.clean.bed <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-k1,1</span> <span class="at">-k2,2n</span> <span class="at">-k3,3n</span>  <span class="op">&gt;</span><span class="va">$workdir</span>/alignment/bed/<span class="va">${sample_name}</span>_bowtie2_rmDup.fragments.bed</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="downsampling">Downsampling<a class="anchor" aria-label="anchor" href="#downsampling"></a>
</h2>
<p>Can be done from fastqs or bam files</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Linux command</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load samtools/1.3.1</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="va">bamdir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY/alignment/bam"</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="va">sample_name</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="va">read_proportion</span><span class="op">=</span><span class="st">".05"</span> <span class="co">#this would be 5%</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="va">seed</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-bs</span> <span class="va">${seed}${read_proportion}</span> <span class="va">$bamdir</span>/{sample_name}_bowtie2.mapped.bam <span class="op">&gt;</span> <span class="va">$bamdir</span>/{sample_name}_subsample.mapped.bam</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="make-normalized-bedgraph">Make (normalized) bedgraph<a class="anchor" aria-label="anchor" href="#make-normalized-bedgraph"></a>
</h2>
<p>Optional as this does not affect peak calling, only bedgraph
values.</p>
<p>*chrom sizes from <a href="http://hgdownload.cse.ucsc.edu/goldenpath/hg19/bigZips/hg19.chrom.sizes" class="external-link uri">http://hgdownload.cse.ucsc.edu/goldenpath/hg19/bigZips/hg19.chrom.sizes</a></p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Linux command</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load bedtools/2.25.0</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY/alignment"</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="va">chromSize</span><span class="op">=</span><span class="st">"/rds/general/user/la420/home/hg19/other/hg19.chrom.sizes"</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="va">sample_name</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="co">#total number of (paired-end) reads, or 2x the number of fragments</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="va">seqDepthDouble</span><span class="op">=</span><span class="kw">`</span><span class="ex">samtools</span> view <span class="at">-F</span> 0x04 <span class="va">$workdir</span>/alignment/sam/<span class="va">${sample_name}</span>_trimmed_bowtie2.sam <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span><span class="kw">`</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="va">seqDepth</span><span class="op">=</span><span class="va">$((seqDepthDouble</span><span class="op">/</span><span class="dv">2</span><span class="va">))</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">$seqDepth</span><span class="st">"</span> <span class="ot">-gt</span> <span class="st">"1"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>    <span class="va">scale_factor</span><span class="op">=</span><span class="kw">`</span><span class="bu">echo</span> <span class="st">"1000000 / </span><span class="va">$seqDepth</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">bc</span> <span class="at">-l</span><span class="kw">`</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"Scaling factor for </span><span class="va">$sample_name</span><span class="st"> is: </span><span class="va">$scale_factor</span><span class="st">"</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>    <span class="ex">bedtools</span> genomecov <span class="at">-bg</span> <span class="at">-scale</span> <span class="va">$scale_factor</span> <span class="at">-i</span> <span class="va">$workdir</span>/bed/<span class="va">${sample_name}</span>_bowtie2_rmDup.fragments.bed <span class="at">-g</span> <span class="va">$chromSize</span> <span class="op">&gt;</span> <span class="va">$workdir</span>/bedgraph/<span class="va">${sample_name}</span>_norm_bowtie2_rmDup.fragments.bedgraph</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a><span class="cf">fi</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="peak-calling-seacr">Peak calling (SEACR)<a class="anchor" aria-label="anchor" href="#peak-calling-seacr"></a>
</h2>
<p>*No IgG control, so using “non”</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Linux command</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load anaconda3/personal</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate seacr_env</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="va">seacr</span><span class="op">=</span><span class="st">"bash </span><span class="va">$HOME</span><span class="st">/anaconda3/envs/seacr_env/bin/SEACR_1.3.sh"</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY"</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a><span class="va">sample_name</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a><span class="va">$seacr</span> <span class="va">$workdir</span>/alignment/bedgraph/<span class="va">${sample_name}</span>_bowtie2_rmDup.fragments.bedgraph 0.01 non stringent <span class="va">$workdir</span>/peakCalling/SEACR/<span class="va">${sample_name}</span>_seacr_top0.01.peaks</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p>Prepare bigwig file:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Linux command</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load anaconda3/personal</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate deepenv</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load samtools/1.3.1</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="va">bamCoverage</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/anaconda3/envs/deepenv/bin/bamCoverage"</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY"</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="va">sample_name</span><span class="op">=</span><span class="st">"Diagenode_S1"</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir $workdir/alignment/bigwig</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> sort <span class="at">-o</span> <span class="va">$workdir</span>/alignment/bam/<span class="va">${sample_name}</span>_rmDup.sorted.bam <span class="va">$workdir</span>/alignment/bam/<span class="va">${sample_name}</span>_bowtie2_rmDup.mapped.bam</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> index <span class="va">$workdir</span>/alignment/bam/<span class="va">${sample_name}</span>_rmDup.sorted.bam</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a><span class="va">$bamCoverage</span> <span class="at">-b</span> <span class="va">$workdir</span>/alignment/bam/<span class="va">${sample_name}</span>_rmDup.sorted.bam <span class="at">-o</span> <span class="va">$workdir</span>/alignment/bigwig/<span class="va">${sample_name}</span>_rmDup_raw.bw       </span></code></pre></div>
<div class="section level3">
<h3 id="heatmap-over-transcription-units">Heatmap over transcription units<a class="anchor" aria-label="anchor" href="#heatmap-over-transcription-units"></a>
</h3>
<p>hg19 genes from: <a href="https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/genes/hg19.ncbiRefSeq.gtf.gz" class="external-link uri">https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/genes/hg19.ncbiRefSeq.gtf.gz</a></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Linux command</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load anaconda3/personal</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate deepenv</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY"</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="va">plotHeatmap</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/anaconda3/envs/deepenv/bin/plotHeatmap"</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="va">computeMatrix</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/anaconda3/envs/deepenv/bin/computeMatrix"</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="va">sample_1</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a><span class="va">sample_2</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a><span class="va">cores</span><span class="op">=</span>8</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir -p $workdir/heatmap</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a><span class="ex">computeMatrix</span> scale-regions <span class="at">-S</span> <span class="va">$workdir</span>/alignment/bigwig/<span class="va">${sample_1}</span>_rmDup_raw.bw <span class="dt">\</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>                               <span class="va">$workdir</span>/alignment/bigwig/<span class="va">${sample_2}</span>_rmDup_raw.bw <span class="dt">\</span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>                              <span class="at">-R</span> <span class="va">$HOME</span>/hg19/other/hg19.ncbiRefSeq.gtf.gz <span class="dt">\</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>                              <span class="at">--beforeRegionStartLength</span> 3000 <span class="dt">\</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>                              <span class="at">--regionBodyLength</span> 5000 <span class="dt">\</span></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>                              <span class="at">--afterRegionStartLength</span> 3000 <span class="dt">\</span></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>                              <span class="at">--skipZeros</span> <span class="dt">\</span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>                              <span class="at">-o</span> <span class="va">$workdir</span>/heatmap/matrix_gene.mat.gz <span class="dt">\</span></span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>                              <span class="at">-p</span> <span class="va">$cores</span></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a><span class="ex">plotHeatmap</span> <span class="at">-m</span> <span class="va">$workdir</span>/heatmap/hg19_gene/matrix_gene.mat.gz <span class="at">-out</span> <span class="va">$workdir</span>/heatmap/heatmap_transcription_units.png <span class="at">--sortUsing</span> sum</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="heatmap-over-cuttag-peaks">Heatmap over CUT&amp;Tag peaks<a class="anchor" aria-label="anchor" href="#heatmap-over-cuttag-peaks"></a>
</h3>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Linux command</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load anaconda3/personal</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate deepenv</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY"</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="va">plotHeatmap</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/anaconda3/envs/deepenv/bin/plotHeatmap"</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="va">computeMatrix</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/anaconda3/envs/deepenv/bin/computeMatrix"</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="va">sample_name</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="va">cores</span><span class="op">=</span>8</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'{split($6, summit, ":"); split(summit[2], region, "-"); print summit[1]"\t"region[1]"\t"region[2]}'</span> <span class="va">$workdir</span>/peakCalling/SEACR/<span class="va">${sample_name}</span>_seacr_top0.01.peaks.stringent.bed <span class="op">&gt;</span><span class="va">$workdir</span>/peakCalling/SEACR/<span class="va">${sample_name}</span>_seacr_top0.01.peaks.summitRegion.bed</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a><span class="ex">computeMatrix</span> reference-point <span class="at">-S</span> <span class="va">$workdir</span>/alignment/bigwig/<span class="va">${sample_name}</span>_rmDup_raw.bw <span class="at">-R</span> <span class="va">$workdir</span>/peakCalling/SEACR/<span class="va">${sample_name}</span>_seacr_top0.01.peaks.summitRegion.bed <span class="at">--skipZeros</span> <span class="at">-o</span> <span class="va">$workdir</span>/peakCalling/SEACR/<span class="va">${sample_name}</span>_SEACR.mat.gz <span class="at">-p</span> <span class="va">$cores</span> <span class="at">-a</span> 3000 <span class="at">-b</span> 3000 <span class="at">--referencePoint</span> center</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a><span class="ex">plotHeatmap</span> <span class="at">-m</span> <span class="va">$workdir</span>/peakCalling/SEACR/<span class="va">${sample_name}</span>_SEACR.mat.gz <span class="at">-out</span> <span class="va">$workdir</span>/peakCalling/SEACR/<span class="va">${sample_name}</span>_SEACR_heatmap.png <span class="at">--sortUsing</span> sum <span class="at">--startLabel</span> <span class="st">"Peak Start"</span> <span class="at">--endLabel</span> <span class="st">"Peak End"</span> <span class="at">--xAxisLabel</span> <span class="st">""</span> <span class="at">--regionsLabel</span> <span class="st">"Peaks"</span> <span class="at">--samplesLabel</span> <span class="st">"</span><span class="va">${sample_name}</span><span class="st">"</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="peak-calling-macs2">Peak calling (MACS2)<a class="anchor" aria-label="anchor" href="#peak-calling-macs2"></a>
</h2>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Linux command</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load anaconda3/personal</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate macs2_env</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY"</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="va">out_dir</span><span class="op">=</span><span class="st">"</span><span class="va">${workdir}</span><span class="st">/peakCalling"</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir -p $outdir</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir -p $outdir/MACS2</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="va">sample_name</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a><span class="ex">macs2</span> callpeak <span class="at">-t</span> <span class="va">$workdir</span>/alignment/bam/<span class="va">${sample_name}</span>_rmDup.sorted.bam <span class="at">-q</span> 0.05 <span class="at">-n</span> <span class="va">${sample_name}</span>_q0.05 <span class="at">-f</span> BAMPE <span class="at">-g</span> hs <span class="at">--keep-dup</span> all <span class="at">--nomodel</span> <span class="at">--outdir</span> <span class="va">$out_dir</span>/MACS2</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="find-motifs">Find motifs<a class="anchor" aria-label="anchor" href="#find-motifs"></a>
</h2>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load anaconda3/personal</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate homerenv</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="st">"/rds/general/project/neurogenomics-lab/live/GitRepos/CUT_n_TAG/processed_data/HKTCKBBXY"</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="va">peakdir</span><span class="op">=</span><span class="st">"</span><span class="va">${workdir}</span><span class="st">/peakCalling"</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="va">outdir</span><span class="op">=</span><span class="st">"</span><span class="va">${workdir}</span><span class="st">/motifs"</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir -p $outdir</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a><span class="va">caller</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a><span class="va">sample_name</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"..."</span></span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a><span class="co"># need separate folder per sample</span></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a><span class="co">#mkdir -p $outdir/${sample_name}</span></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a><span class="ex">findMotifsGenome.pl</span> <span class="va">$peakdir</span>/<span class="va">${caller}</span>/<span class="va">${sample_name}</span>_<span class="va">${caller}</span>.bed <span class="va">$genome</span> <span class="va">$out_dir</span> <span class="at">-size</span> given</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="genome-wide-correlation">Genome-wide correlation<a class="anchor" aria-label="anchor" href="#genome-wide-correlation"></a>
</h2>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Linux command</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -l walltime=72:00:00</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -l select=1:ncpus=32:mem=124gb</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load anaconda3/personal</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate bedtools_env</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a><span class="va">abdir</span><span class="op">=</span><span class="st">"/rds/general/user/la420/home/CUTnTAG/antibodies"</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a><span class="va">corrdir</span><span class="op">=</span><span class="st">"</span><span class="va">${abdir}</span><span class="st">/correlation"</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="va">bamdir</span><span class="op">=</span><span class="st">"</span><span class="va">${abdir}</span><span class="st">/bam/sorted_bam"</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a><span class="va">beddir</span><span class="op">=</span><span class="st">"</span><span class="va">${corrdir}</span><span class="st">/ENCODE_peak_files"</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a><span class="va">ENCODE_bamdir</span><span class="op">=</span><span class="st">"</span><span class="va">${corrdir}</span><span class="st">/ENCODE_peak_files/bam/sorted_bam/fixed"</span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a><span class="va">old_ENCODE_bamdir</span><span class="op">=</span><span class="st">"</span><span class="va">${corrdir}</span><span class="st">/ENCODE_peak_files/bam/sorted_bam"</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a><span class="va">outdir</span><span class="op">=</span><span class="st">"</span><span class="va">${corrdir}</span><span class="st">/all_marks"</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a><span class="va">mergedbeddir</span><span class="op">=</span><span class="st">"</span><span class="va">${abdir}</span><span class="st">/all_peaks/merged"</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a><span class="va">hg19_dir</span><span class="op">=</span><span class="st">"/rds/general/user/la420/home/hg19/other"</span></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a><span class="co">#mark="H3K27ac_ENCFF044JNJ"</span></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a><span class="co">#mark="H3K27me3_ENCFF126QYP"</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a><span class="ex">bedtools</span> multicov <span class="at">-bams</span> <span class="dt">\</span></span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a><span class="va">$bamdir</span>/Abcam-ab177178_rmDup.sorted.fixed2.bam <span class="dt">\</span></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a><span class="va">$bamdir</span>/Abcam-ab4729_rmDup.sorted.fixed2.bam <span class="dt">\</span></span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a><span class="va">$bamdir</span>/ActiveMotif_rmDup.sorted.fixed2.bam <span class="dt">\</span></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a><span class="va">$bamdir</span>/Diagenode_100x_rmDup.sorted.fixed2.bam <span class="dt">\</span></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a><span class="va">$bamdir</span>/Diagenode_50x_rmDup.sorted.fixed2.bam <span class="dt">\</span></span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a><span class="va">$bamdir</span>/H3K27me3_rmDup.sorted.fixed2.bam <span class="dt">\</span></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a><span class="va">$bamdir</span>/H3K27ac_SRR8383507_rmDup.sorted.fixed2.bam <span class="dt">\</span></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a><span class="va">$bamdir</span>/H3K27ac_SRR8383508_rmDup.sorted.fixed2.bam <span class="dt">\</span></span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a><span class="va">$bamdir</span>/H3K27me3_SRR11074238_rmDup.sorted.fixed2.bam <span class="dt">\</span></span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a><span class="va">$bamdir</span>/H3K27me3_SRR11074239_rmDup.sorted.fixed2.bam <span class="dt">\</span></span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a><span class="va">$bamdir</span>/ActiveMotif_H3K27ac_rmDup.sorted.fixed2.bam <span class="dt">\</span></span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a><span class="va">$bamdir</span>/H3K27ac_SRR8581604_CnR_rmDup.sorted.fixed2.bam <span class="dt">\</span></span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a><span class="va">$bamdir</span>/H3K27me3_SRR907370_CnR_rmDup.sorted.fixed2.bam <span class="dt">\</span></span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a><span class="va">$old_ENCODE_bamdir</span>/H3K27ac_ENCFF384ZZM_sorted.bam <span class="dt">\</span></span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a><span class="va">$old_ENCODE_bamdir</span>/H3K27me3_ENCFF676ORH_sorted.bam <span class="dt">\</span></span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a><span class="va">$ENCODE_bamdir</span>/H3K4me1_ENCFF196QGZ_sorted.fixed.bam <span class="dt">\</span></span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a><span class="va">$ENCODE_bamdir</span>/H3K4me3_ENCFF915MJO_sorted.fixed.bam <span class="dt">\</span></span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a><span class="va">$ENCODE_bamdir</span>/H3K9ac_ENCFF204DNG_sorted.fixed.bam <span class="dt">\</span></span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a><span class="va">$ENCODE_bamdir</span>/H3K9me3_ENCFF805FLY_sorted.fixed.bam <span class="dt">\</span></span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a><span class="va">$ENCODE_bamdir</span>/H3K36me3_ENCFF190AQC_sorted.fixed.bam <span class="dt">\</span></span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a><span class="va">$ENCODE_bamdir</span>/DNase_ENCFF826DJP_sorted.fixed.bam <span class="dt">\</span></span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a>-bed <span class="va">$hg19_dir</span>/hg19_500bp_windows.bed <span class="op">&gt;</span> <span class="va">$outdir</span>/hg19_500bp_overlaps.txt</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R Under development (unstable) (2022-03-03 r81847)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.4 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] EpiProcess_0.99.0 BiocStyle_2.23.1 </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] Rcpp_1.0.8          knitr_1.37          magrittr_2.0.2     </span>
<span class="co">##  [4] rappdirs_0.3.3      lattice_0.20-45     R6_2.5.1           </span>
<span class="co">##  [7] ragg_1.2.2          rlang_1.0.2         fastmap_1.1.0      </span>
<span class="co">## [10] stringr_1.4.0       tools_4.2.0         grid_4.2.0         </span>
<span class="co">## [13] data.table_1.14.2   xfun_0.30           png_0.1-7          </span>
<span class="co">## [16] cli_3.2.0           jquerylib_0.1.4     systemfonts_1.0.4  </span>
<span class="co">## [19] htmltools_0.5.2     yaml_2.3.5          digest_0.6.29      </span>
<span class="co">## [22] rprojroot_2.0.2     pkgdown_2.0.2.9000  bookdown_0.24      </span>
<span class="co">## [25] textshaping_0.3.6   Matrix_1.4-0        BiocManager_1.30.16</span>
<span class="co">## [28] purrr_0.3.4         sass_0.4.0          fs_1.5.2           </span>
<span class="co">## [31] memoise_2.0.1       cachem_1.0.6        evaluate_0.15      </span>
<span class="co">## [34] rmarkdown_2.12      echoconda_0.99.5    stringi_1.7.6      </span>
<span class="co">## [37] compiler_4.2.0      bslib_0.3.1         desc_1.4.1         </span>
<span class="co">## [40] reticulate_1.24     jsonlite_1.8.0</span></code></pre>
</details><p><br></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
